core/src/main/scala/org/broadinstitute/dsde/rawls/entities/base/EntityProvider.scala

```diff
@@ -26,16 +25,31 @@ import scala.util.Try
  * trait definition for entity providers.
  */
 trait EntityProvider {
+  // entityStoreId is used by subclasses to identify themselves
   def entityStoreId: Option[String]
 
-  def entityTypeMetadata(useCache: Boolean): Future[Map[String, EntityTypeMetadata]]
+  // ----- implementation methods follow:
+
+  def batchUpdateEntities(entityUpdates: Seq[EntityUpdateDefinition]): Future[Traversable[Entity]]
+
+  def batchUpsertEntities(entityUpdates: Seq[EntityUpdateDefinition]): Future[Traversable[Entity]]
+
+  def copyEntities(sourceWorkspaceContext: Workspace,
+                   destWorkspaceContext: Workspace,
+                   entityType: String,
+                   entityNames: Seq[String],
+                   linkExistingEntities: Boolean,
+                   parentContext: RawlsRequestContext
+  ): Future[EntityCopyResponse]
```

Comment:
I've alphabetized the methods in this class (except for `entityStoreId` at the top) - apologies for the extra diffs